salt/utils/win_network.py

```diff
@@ -0,0 +1,337 @@
+            int_dict.setdefault(
+                names[addrs.Address.AddressFamily], []).append(ip_info)
+    return int_dict
+
+
+def _get_ip_gateway_info(i_face):
+    ip_properties = i_face.GetIPProperties()
+    int_dict = {}
+    if ip_properties.GatewayAddresses.Count > 0:
+        names = {af_inet: 'ip_gateways',
+                 af_inet6: 'ipv6_gateways'}
+        for addrs in ip_properties.GatewayAddresses:
+            int_dict.setdefault(
+                names[addrs.Address.AddressFamily],
+                []).append(addrs.Address.ToString().split('%')[0])
+    return int_dict
+
+
+def _get_ip_dns_info(i_face):
+    ip_properties = i_face.GetIPProperties()
+    int_dict = {}
+    if ip_properties.DnsAddresses.Count > 0:
+        names = {af_inet: 'ip_dns',
+                 af_inet6: 'ipv6_dns'}
+        for addrs in ip_properties.DnsAddresses:
+            int_dict.setdefault(
+                names[addrs.AddressFamily],
+                []).append(addrs.ToString().split('%')[0])
+    return int_dict
+
+
+def _get_ip_multicast_info(i_face):
+    ip_properties = i_face.GetIPProperties()
+    int_dict = {}
+    if ip_properties.MulticastAddresses.Count > 0:
+        names = {af_inet: 'ip_multicast',
+                 af_inet6: 'ipv6_multicast'}
+        for addrs in ip_properties.MulticastAddresses:
+            int_dict.setdefault(
+                names[addrs.Address.AddressFamily],
+                []).append(addrs.Address.ToString().split('%')[0])
+    return int_dict
+
+
+def _get_ip_anycast_info(i_face):
+    ip_properties = i_face.GetIPProperties()
+    int_dict = {}
+    if ip_properties.AnycastAddresses.Count > 0:
+        names = {af_inet: 'ip_anycast',
+                 af_inet6: 'ipv6_anycast'}
+        for addrs in ip_properties.AnycastAddresses:
+            int_dict.setdefault(
+                names[addrs.Address.AddressFamily],
+                []).append(addrs.Address.ToString())
+    return int_dict
+
+
+def _get_ip_wins_info(i_face):
+    ip_properties = i_face.GetIPProperties()
+    int_dict = {}
+    if ip_properties.WinsServersAddresses.Count > 0:
+        for addrs in ip_properties.WinsServersAddresses:
+            int_dict.setdefault(
+                'ip_wins', []).append(addrs.ToString())
+    return int_dict
+
+
+def _get_network_interfaces():
+    clr.AddReference('System.Net')
+    return NetworkInformation.NetworkInterface.GetAllNetworkInterfaces()
+
+
+def get_interface_info_dot_net():
+    '''
+    Uses .NET 4.0+ to gather Network Interface information. Should only run on
+    Windows systems newer than Windows 7/Server 2008R2
+    Returns:
+        dict: A dictionary of information about all interfaces on the system
+    '''
+    clr.AddReference('System.Net')
+    interfaces = _get_network_interfaces()
+
+    int_dict = {}
+    for i_face in interfaces:
+        int_dict[i_face.Name] = _get_base_properties(i_face)
+        int_dict[i_face.Name].update(_get_ip_base_properties(i_face))
```

Comment:
doing int_dict[i_face.Name] on all following lines are doing the lookups each time.

Maybe it's worth having a separate dictionary and set it as int_dict[i_face.Name] once, at the end.